aws/resource_aws_waf_web_acl.go

```diff
@@ -64,6 +64,18 @@ func resourceAwsWafWebAcl() *schema.Resource {
 							Type:     schema.TypeInt,
 							Required: true,
 						},
+						"type": &schema.Schema{
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+								value := v.(string)
+								if value != waf.WafRuleTypeRegular && value != waf.WafRuleTypeRateBased {
+									errors = append(errors, fmt.Errorf(
+										"%q must be one of %s | %s", k, waf.WafRuleTypeRegular, waf.WafRuleTypeRateBased))
+								}
+								return
+							},
+						},
```

Comment:
Is there any particular reason why this change should be included in this PR? How is it related to WAF Rate Based Rule?

Reply:
I mentioned this in https://github.com/terraform-providers/terraform-provider-aws/pull/1285. By default, WebACL works with `REGULAR` rules and you will get an error if you pass `rule_id` of a `RATE_BASED` rule. This is documented in https://github.com/aws/aws-sdk-go/blob/master/service/waf/api.go#L5708-L5713

Reply:
I see, but I think we can't just introduce a `Required` field - that would break it for users with existing configs which don't have this field there.
This applies to your other change in `aws/resource_aws_waf_rule.go` too.

Reply:
Do you mind changing it to Optional?

Reply:
btw. you might be able to avoid changing the `updateWebAclResource` if you also add the current default value as `Default` - that way it should just work the same way as before, yet allow users to specify `type = "RATE_BASED"` if they need to.

Reply:
@radeksimko Thank you for the review. I believe the `Required` change for `aws/resource_aws_waf_rule.go` is fine, as we just duplicate AWS validation, it's not a breaking change.
I made `type` optional and default to `REGULAR` as recommended.

Reply:
You're right - just checked the [API docs](http://docs.aws.amazon.com/waf/latest/APIReference/API_Predicate.html) - it was probably just overlooked.

Thanks.